Homeworks/05/src/App.tsx

```diff
@@ -1,5 +1,135 @@
+        console.error(err)
+      })
+    }
+
+    const onScroll = () => {
+      if (document.body.scrollHeight - 200 <= window.scrollY + window.innerHeight) {
+        setCount(count => count + 1)
+      }
+    }
+      
+    window.addEventListener('scroll', onScroll, { passive: true })
+    return () => window.removeEventListener('scroll', onScroll)
+  }, [count])
+
+
+  return (
+    <Flex direction="column" fontFamily="'Poppins', sans-serif" data-theme={theme}>
+      <Favorites 
+      theme={theme}
+      isOpen={open}
+      onClose={() => setOpen(false)}
+      pokemons={favorites}
+      addFavorite={(e: Pokemon) => setFavorites(favorites => [...favorites, e])} 
+      removeFavorite={(e: string) => setFavorites(favorites => favorites.filter(fav => fav.name !== e))} 
+      ></Favorites>
+      <Flex 
+      h={{ base: "3xl", lg: "s1" }}
+      w="100%"
+      background="background.primary" 
+      position="sticky" 
+      top="0" 
+      justifyContent="space-between" 
+      alignItems="center"
+      zIndex="1"
+      paddingX={{ base: "sm", sm: "xl"}}
+      paddingY={{ base: "sm", sm: "xl"}}>
+        <styled.img src={`src/images/heart.png`} 
+        h={{ base: "lg", sm: "xl"}} 
+        scale="fit" 
+        cursor={{_hover: "pointer"}}
+        animationDirection="alternate"
+        animation={{_hover: "grow 0.6s infinite"}}
+        onClick={() => setOpen(true)}
+        />
+        <Flex 
+        color="text.primary" 
+        textStyle={{ base: "display.small", sm: "display.medium" }} 
+        justifyContent="center" alignItems="center" 
+        h={{ base: "lg", sm: "xl" }}>
+          <styled.img src={`src/images/poke-ball.png`} h={{ base: "lg", sm: "xl" }} scale="fit"/>
+          POKEDEX
+        </Flex>
+        <styled.img src={`src/images/gear.png`} 
+        h={{ base: "lg", sm: "xl" }} 
+        scale="fit"
+        animation={{_hover: "spin 2s linear infinite"}}
+        cursor={{_hover: "pointer"}}
+        onClick={() => setThemeBox(themeBox === "none"? "flex": "none")}
+        />
+      </Flex>
+      <Flex direction="column">
+        <Flex
+        display={themeBox}
+        color="text.default"
+        direction="column"
+        position="fixed"
+        right="0"
+        top={{ base: "3xl", lg: "s1" }}
+        margin="md"
+        justifyContent="center"
+        zIndex="1"
+        bg="background.details"
+        border="solid"
+        h="100px"
+        w="200px"
+        borderRadius="md"
+        onChange={e => setTheme(e.target.value)}>
+          <styled.span marginLeft="sm">Theme:</styled.span>
+          <Flex alignItems="center" h="2xl">
+            <styled.input type="radio" value="base" name="theme" margin="sm" defaultChecked/>
+            <styled.label>Light</styled.label>
+          </Flex>
+          <Flex alignItems="center" h="2xl">
+            <styled.input type="radio" value="dark" name="theme" margin="sm"/>
+            <styled.label>Dark</styled.label>
```

Comment:
`styled` tag nema potrebe koristit ako ne stiliziramo element
```suggestion
            <label>Dark</label>
```